### PR TITLE
add capability for decoding of omps ozone products

### LINKED ISSUE #1591: create a converter for ozone profile product
## Description

this converter will decode what is current a netCDF4 based product of Ozone profile from the OMPS sensor for the nadir profiling component

## Requirements

input files, at this time the V8PRO product is what is being used as the initial contribution

## Acceptance Criteria (Definition of Done)

a IODA formatted file following current conventions

## Dependencies

python bindings to IODA

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

this adds a capability to decode an OMPS netCDF4 file containing the nadir profiling NOAA product

V8PRO EDR (Environmental Data Record)

## Issue(s) addressed

Resolves #1591 

## Impact

Capability to convert to IODA from raw product file

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

File: src/hdf5/ozone_product_2ioda.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2020-2023 UCAR
```
Comment (BenjaminRuston) [inline src/hdf5/ozone_product_2ioda.py]:
```suggestion
# (C) Copyright 2020-2024 UCAR
```

should likely update

**Comment by fcvdb** [comment]
From [STAR JPSS - Environmental Data Record Teams - OMPS Ozone EDRs](https://www.star.nesdis.noaa.gov/jpss/ozone.php). NESDIS produces two files, the Total Ozone (V8TOz) and the Profile (V8Pro). 
 

From WMO OSCAR: WMO OSCAR  |  Details for Instrument OMPS-nadir
Mapper: cross-track swath 2800 km, along-track one 50-km strip in 7.6 s. Nadir profiler: one along-track sounding every 38 s (250 km)
Mapper: 50 km on S-NPP but improved to 10km on NOAA-21.  Nadir profiler: 250 km
Global coverage: mapper once/day, nadir profiler in 12 days. Daylight only

Each NESDIS Nadir profiler file V8E PRO EDR, like the one added in the PR `test/testinput/V8PRO-EDR_v3r3_j01_s202202142345436_e202202142346211_c202202150032490.nc` contains profiles on 21 layers of Ozone (O3FINAL) in DU units, as well as its integrated value over the column (ColumnAmountO3_Profile) in DU as well. The profiles are provided on a 5x5 grid (ntimes x nlfov). It seems, however, that only profiles for nlfov=1 are present, so there are between 1 to 5 profiles per files. The ColumnAmountO3_Profile is the sum of the 21 layers present in the O3FINAL profile.

        nTimes = 5 ;
        nIfov = 5 ;
        nLayers = 21 ;


        float O3FINAL(nTimes, nIfov, nLayers) ;
                O3FINAL:long_name = "Ozone Solution Profile" ;
                O3FINAL:units = "Dobson Units" ;
                O3FINAL:parameter_type = "V8PRO data" ;
                O3FINAL:coordinates = "Time Longitude Latitude" ;
                O3FINAL:_FillValue = -9999.f ;
                O3FINAL:valid_range = 0, 1000 ;

        float ColumnAmountO3_Profile(nTimes, nIfov) ;
                ColumnAmountO3_Profile:long_name = "Ozone Profile Column Sum" ;
                ColumnAmountO3_Profile:units = "Dobson Units" ;
                ColumnAmountO3_Profile:parameter_type = "V8PRO data" ;
                ColumnAmountO3_Profile:coordinates = "Time Longitude Latitude" ;
                ColumnAmountO3_Profile:_FillValue = -9999.f ;
                ColumnAmountO3_Profile:valid_range = 10, 1000 ;

**Comment by fcvdb** [comment]
According to the IODA [conventions](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/conventions/tbls.html) ozone should be converted into mol m-2. A conversion factor of `4.4615E-04` must be applied to convert DU into mol -m2.

**Review by fcvdb — APPROVED** [review]
I tested the converter on recently downloaded  NESDIS TOZ file. Thanks Ben!

**Comment by nancylbaker** [comment]
I have confirmed these details, i.e. the input files,  descriptions for the two products and unit conversions.  Thanks Ben and Francois.

**Review by nancylbaker — APPROVED** [review]

**Comment by jeromebarre** [comment]
Why hasn't this been added to compo or compo made aware of this? There are already converters for this there...

**Pull request merged**